src/execution_engines/test/security/node_js/page_boundaries.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { expect } from 'chai'
+import NodeEngine from '../../../node_js'
+import { ExecutionEngines } from '../../../ExecutionEngine'
+
+describe('Puppeteer Page Boundaries', () => {
+  beforeEach(() => {
+    process.env.ENGINE = ExecutionEngines.nodejs
+  })
+
+  it('state set on the window by one contract cannot be read by another', async () => {
```

Comment:
This could be a `describe('State isolation`, then move the setup and teardown into before/after